ending_location_visitor.py

```diff
@@ -1,32 +1,223 @@
+        astroid's `node_class`. Internally, appends the function to a
+        dictionary in the TransformVisitor class. The keys are the node types,
+        each node must have been assigned/registered a transform function!!
+        The transform function may return a value which is then used to
+        substitute the original node in the tree.
+        params: node_class, transform function
+
+    We store the correct values as: 
+        [lineno, end_lineno, col_offset, end_col_offset]
+    """
+
+    @classmethod
+    def setUpClass(self):
+        '''A class method called before tests in an individual class run.
+        setUpClass is called with the class as the only argument and must be
+        decorated as a classmethod():'''
+        
+        # A visitor to transform the nodes.
+        self.ending_transformer = EasyTransformVisitor()  # TransformVisitor
+        # A visitor to test the nodes property correctness.
+        self.ending_visitor = EndingVisitor()
+        # Register all `transform(node)` functions here...
+        # self.ending_transformer.register_transform(astroid.Const, set_const)
+        # self.ending_transformer.register_transform(astroid.BinOp, set_binop)
+        self.ending_transformer.easy_register_transform('Const')
+        self.ending_transformer.easy_register_transform('BinOp')
+        
+
+    def setUp(self):
+        '''Method called to prepare the test fixture. This is called immediately
+        before calling the test method; other than AssertionError or SkipTest,
+        any exception raised by this method will be considered an error rather
+        than a test failure. The default implementation does nothing.'''
+        pass
+
+    def tearDown(self):
+        '''Method called immediately after the test method has been called and
+        the result recorded. This is called even if the test method raised an
+        exception, so the implementation in subclasses may need to be
+        particularly careful about checking internal state. Any exception,
+        other than AssertionError or SkipTest, raised by this method will be
+        considered an additional error rather than a test failure (thus
+        increasing the total number of reported errors). This method will only
+        be called if the setUp() succeeds, regardless of the outcome of the
+        test method. The default implementation does nothing.'''
+        self.ending_visitor.reset()
+
+    def _get_file_as_module(self, file_location):
+        """Given a filepath (file_location), parse with astroid, and return 
+        the module.
+        """
+        with open(file_location) as f:
+            content = f.read()
+        return self._get_string_as_module(content)
+
+    def _get_string_as_module(self, string):
+        """Parse the string with astroid, and return the module.
+        """
+        return astroid.parse(string)
+
+    def _are_equal(self, expected, props_assigned):
```

Comment:
assertListEqual seems to care about order of the items, which i would like to try to avoid.